### PR TITLE
Handle markdown in control and systems implementations

### PR DESCRIPTION
**Pull request by kperk103**
This will allow control implementation descriptions and
system implementation remarks to handle markdown so that
the text appears in a clean and readable format.

Part of #156.

### CODE REVIEW

File: example/package.json
Comment (tuckerzp) [inline example/package.json:48]: I think this change may already be present in `develop`

Comment (Bronstrom) [reply example/package.json:48]: Is this dependency needed in `example`? From what I understood from the issue, we're altering `src/component`s with markdown. Are we altering anything with markdown in `example`?

Comment (kperk103) [reply example/package.json:48]: We aren't altering anything with markdown in `example`, however in one of my previous commits (d17d0a4), the following error is displayed when trying to build and test:
Screenshot (10)
This is the same error I was getting when trying to run the component tests locally before adding the dependency in `package.json`. Adding the same dependency in `example/package.json` fixed the error as well. You are correct that nothing is being altered with markdown in `example`, so I am confused as to why this is necessary.

**Review by Bronstrom — APPROVED** [review]
Good work - looks good to me!

File: src/components/OSCALControlProse.js
Comment (tuckerzp) [inline src/components/OSCALControlProse.js:388]: To stay consistent with discussions from #548 on markdown vs markup, can we change this to:
```suggestion
  const statementByComponentDescriptionMarkup = (
```

Comment (kperk103) [reply src/components/OSCALControlProse.js]: That makes sense! My most recent commit has this change

**Comment by kperk103** [comment]
@tuckerzp Here are some examples of markdown being handled in control implementation descriptions and system implementation remarks:
Screenshot (13)
Screenshot (15)

**Comment by kylelaker** [comment]
> @tuckerzp Here are some examples of markdown being handled in control implementation descriptions and system implementation remarks: Screenshot (13) Screenshot (15)

Wait no way! This is so cool! It works in the tooltips too?! I literally love this.

**Review by kylelaker — CHANGES_REQUESTED** [review]
Still need to request changes though; sorry. Looks like maybe putting the Prose inside a `<Typography>` causes new console warnings. @tuckerzp can you support that?

**Comment by tuckerzp** [comment]
> new console warnings

After all my fuss about warnings, I am embarrassed to have missed that! I will assist in fixing them. 

https://github.com/EasyDynamics/oscal-react-library/actions/runs/2825466728

**Comment by kylelaker** [comment]
> Looks like maybe putting the Prose inside a `<Typography>` causes new console warnings

I think if we need to (if we can't just always tell it to use `Typography`) we can pass something (a `Typography`) for `paragraph` or whatever as a prop to the underlying component, instead of wrapping things at a higher level.

File: src/components/OSCALSystemImplementation.js
Comment (tuckerzp) [inline src/components/OSCALSystemImplementation.js:50]: Is this `Typography` even needed?

**Comment by kperk103** [comment]
> Still need to request changes though; sorry. Looks like maybe putting the Prose inside a `<Typography>` causes new console warnings. @tuckerzp can you support that?

I believe that if I use the `OSCALMarkupLine` component from https://github.com/EasyDynamics/oscal-react-library/pull/548/files#diff-09fcaba6157b2ce435ef376c32a2e3b1e705969d23bbc7ebeebf6288c2738786 it gets rid of these console warnings. I can wait until that PR is merged to make these changes if you believe that is the best way to get rid of these warnings.

File: src/components/OSCALSystemImplementation.js
Comment (kperk103) [reply src/components/OSCALSystemImplementation.js:50]: Without the `Typography`, the font of the system implementation remarks changes so it is not consistent with the rest of the page. The first screenshot shows it with the `Typography` and the second screenshot shows it without the `Typography`:
Screenshot (18)
Screenshot (16)

**Comment by tuckerzp** [comment]
> I believe that if I use the `OSCALMarkupLine`...  it gets rid of these console warnings

I agree that would get rid of the console warnings, but am not completely sure if we want these things to be a single line. I would argue that the `description` would make sense to be a `OSCALMarkUpMultiLine`. 

> we can pass something (a `Typography`) for paragraph or whatever as a prop to the underlying component, instead of wrapping things at a higher level.

@kylelaker I may misunderstand what you mean by this, but would doing something similar to [here](https://github.com/EasyDynamics/oscal-react-library/pull/543#discussion_r940488448) be a potential way forward?

I am just not entirely convinced that everywhere we get this warning switching to `OSCALMarkupLine` is the best solution and think our answer may need to be more robust.

**Comment by kylelaker** [comment]
> but would doing something similar to [here](https://github.com/EasyDynamics/oscal-react-library/pull/543#discussion_r940488448) be a potential way forward

I _think_ that's what I'm saying, yeah.

**Comment by kperk103** [comment]
> I agree that would get rid of the console warnings, but am not completely sure if we want these things to be a single line. I would argue that the `description` would make sense to be a `OSCALMarkUpMultiLine`.

If I have a multiline `description` in the control implementation and use `OSCALMarkupLine`, it still works:

Screenshot (19)

The same also works for a multiline `remark` in system implementations.

If you still think it would be best for me to do something similar to the link you provided, I can look into that as well.

**Comment by tuckerzp** [comment]
> If I have a multiline description in the control implementation and use `OSCALMarkupLine`, it still works:

@schen2166 & @kylelaker I am curious on your thoughts about this. It seems that maybe the `OSCALMarkUpMultiLine` is not needed for this current issue? I am unsure if we need to necessarily have both of them now when viewing at least only requires what we have defined as `OSCALMarkupLine`. I have not seen a place where `OSCALMarkUpMultiLine` is strictly needed over `OSCALMarkupLine`.

**Comment by schen2166** [comment]
>  I am curious on your thoughts about this. It seems that maybe the OSCALMarkUpMultiLine is not needed for this current issue?
For the scope of this issue on resolving markdown in prose, I don't see an immediate need to support multiline markdown however it may be something that should be supported in the future. in my opinion, getting `MarkupMultiline` working doesn't need to be a priority. `MarkupLine` should have be able to resolve the issues related to this task

**Comment by kperk103** [comment]
> `MarkupLine` should have be able to resolve the issues related to this task

I agree that this can be used to resolve the issue while still allowing functionality for multiple lines. If this is the solution we choose, I think I will need to wait until https://github.com/EasyDynamics/oscal-react-library/pull/548/files#diff-09fcaba6157b2ce435ef376c32a2e3b1e705969d23bbc7ebeebf6288c2738786 is merged in order to use this component.

**Comment by kylelaker** [comment]
> It seems that maybe the `OSCALMarkUpMultiLine` is not needed for this current issue? I am unsure if we need to necessarily have both of them now when viewing at least only requires what we have defined as `OSCALMarkupLine`. I have not seen a place where `OSCALMarkUpMultiLine` is strictly needed over `OSCALMarkupLine`.

Which one does the spec say that this is?

**Comment by kperk103** [comment]
> Which one does the spec say that this is?

If I'm understanding it correctly, I think the spec says it is a markup-multiline

File: src/components/OSCALMarkupProse.js
Comment (tuckerzp) [inline src/components/OSCALMarkupProse.js:15]: I am worried what this does if `props.paragraphComponent` is `undefined`. I am unsure of the functionality of rewriting components if given a `undefined` value. We may want to put some kind of check. 
```suggestion
          p: props.paragraphComponent ? props.paragraphComponent :  p ,
```


I also think this might cause the `StyledToolTip` to have issues.

Comment (kylelaker) [reply src/components/OSCALMarkupProse.js:15]: `p: props.paragraphComponent ?? 'p'`

Comment (kylelaker) [reply src/components/OSCALMarkupProse.js:15]: Iff we need a check in the first place

Comment (kperk103) [reply src/components/OSCALMarkupProse.js]: Thanks! My most recent commits address this

File: src/components/OSCALControlProse.js
Comment (kylelaker) [inline src/components/OSCALControlProse.js:393]: Should we do this in the case where `statementByComponentDescription` is `null`?

```tsx
const statementByComponentDescriptionMarkup = statementByComponentDescription ? (
  <OSCALMarkupMultiLine
    text={statementByComponentDescription}
    paragraphComponent="p"
  />
);
```

**Review by kylelaker — CHANGES_REQUESTED** [review]

File: src/components/OSCALControlProse.js
Comment (kperk103) [reply src/components/OSCALControlProse.js]: That's a great idea, thank you!

File: src/components/OSCALControlProse.js
Comment (tuckerzp) [inline src/components/OSCALControlProse.js:392]: I thinking mapping `p -> p` is a bit weird here. This goes back to my original comment suggesting the checking for `paragraphComponent` to be `null`. I don't think we should force a rewrite of the `p` tag.

**Review by tuckerzp — APPROVED** [review]

**Review by kylelaker — APPROVED** [review]
Thank you for all your hard work on this and for going back and forth on so many items. Great stuff!

**Pull request merged**
